### PR TITLE
Add details about pull requests to contributing guide

### PR DESCRIPTION
**Pull request by akhenry**
Provides additional information about pull requests to our contributing guide. Addresses #3021 

### Author Checklist
1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? N/A - No code change
3. Command line build passes? Y
4. Changes have been smoke-tested? N/A - No code change
5. Testing instructions included? N/A - No code change

### CODE REVIEW

**Comment by akhenry** [comment]
@charlesh88 Hold off, want to add some more changes to this PR. Inbound...

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -292,10 +298,12 @@ checklist).
...
 4. Changes have been smoke-tested?
+5. Testing instructions included?
 
 ### Reviewer Checklist
 
 1. Changes appear to address issue?
 2. Appropriate unit tests included?
 3. Code style and in-line documentation are appropriate?
 4. Commit messages meet standards?
+5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue)
```
Comment (charlesh88) [inline CONTRIBUTING.md:309]: Suggest rewording this: "If this PR closes an associated issue, has that issue been labeled `unverified`?

**Comment by JoelMcKinnon** [comment]
Looks good to me

File: CONTRIBUTING.md (hunk 1 above)
Comment (akhenry) [reply CONTRIBUTING.md:309]: Hehe, I actually had it worded that way originally. I decided it was too verbose if you're scanning over it every time you file a PR though. I also figured that the instructions about labelling the issue as unverified were more important than the qualifier?

Comment (JoelMcKinnon) [reply CONTRIBUTING.md:309]: Could shorten to:
Associated issued labelled `unverified`? (if PR closes issue)

Comment (shefalijoshi) [reply CONTRIBUTING.md:309]: I'm OK with what Andrew has.

**Comment by akhenry** [comment]
Have updated based on testing discussion on that took place 05/13/2020
